# Evaluation 

## Q1 - Expliquer ce qu'est un KPI ?
Un KPI (Key Performance Indicator) ou ICP (Indicateur Clé de Performance) permet de déterminer les facteurs les plus efficients en mesurant l'efficacite du dispositif suivi.

## Q2 - Importer les librairies pandas et numpy

In [1]:
import pandas as pds
import numpy as np

## Q3 Importer et le dataset "eval.csv"

In [74]:
path = "eval_data.csv"

df = pds.read_csv(path)

## Q4 Afficher les 10 premières lignes du dataset

In [3]:
df.head(10)

Unnamed: 0  order_id  quantity                              item_name  \
0           0         1         1           Chips and Fresh Tomato Salsa   
1           1         1         1                                   Izze   
2           2         1         1                       Nantucket Nectar   
3           3         1         1  Chips and Tomatillo-Green Chili Salsa   
4           4         2         2                           Chicken Bowl   
5           5         3         1                           Chicken Bowl   
6           6         3         1                          Side of Chips   
7           7         4         1                          Steak Burrito   
8           8         4         1                       Steak Soft Tacos   
9           9         5         1                          Steak Burrito   

                                  choice_description item_price  
0                                                NaN        NaN  
1                                       [Clementine]     $3.39   
2                                            [Apple]     $3.39   
3                                                NaN        NaN  
4  [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98   
5  [Fresh Tomato Salsa (Mild), [Rice, Cheese, Sou...    $10.98   
6                                                NaN     $1.69   
7  [Tomatillo Red Chili Salsa, [Fajita Vegetables...    $11.75   
8  [Tomatillo Green Chili Salsa, [Pinto Beans, Ch...     $9.25   
9  [Fresh Tomato Salsa, [Rice, Black Beans, Pinto...     $9.25

## Q5 Afficher la dimension du dataset ?

In [4]:
df.shape

(4622, 6)

## Q6 Afficher, et décrire les colonnes de notre dataset

In [9]:
df.dtypes

Unnamed: 0             int64
order_id               int64
quantity               int64
item_name             object
choice_description    object
item_price            object
dtype: object

## Q7 Afficher les statistiques descriptives du dataset

In [7]:
df.describe()

Unnamed: 0     order_id     quantity
count  4622.000000  4622.000000  4622.000000
mean   2310.500000   927.254868     1.075725
std    1334.400802   528.890796     0.410186
min       0.000000     1.000000     1.000000
25%    1155.250000   477.250000     1.000000
50%    2310.500000   926.000000     1.000000
75%    3465.750000  1393.000000     1.000000
max    4621.000000  1834.000000    15.000000

## Q8 Quel est l'item le plus commandé ?

In [95]:
df.groupby("item_name")["quantity"].sum()

item_name
6 Pack Soft Drink                         55
Barbacoa Bowl                             66
Barbacoa Burrito                          91
Barbacoa Crispy Tacos                     12
Barbacoa Salad Bowl                       10
Barbacoa Soft Tacos                       25
Bottled Water                            211
Bowl                                       4
Burrito                                    6
Canned Soda                              126
Canned Soft Drink                        351
Carnitas Bowl                             71
Carnitas Burrito                          60
Carnitas Crispy Tacos                      8
Carnitas Salad                             1
Carnitas Salad Bowl                        6
Carnitas Soft Tacos                       40
Chicken Bowl                             761
Chicken Burrito                          591
Chicken Crispy Tacos                      50
Chicken Salad                              9
Chicken Salad Bowl                       123


L'item le plus commandé est donc "Chicken Bowl"

## Q9 Combien d'items ont été commandés au total ?


In [52]:
df.quantity.sum()

4972

4972 items ont été commandés au total

## Q10 Combien de chiffre d'affaire à réalisé la société ?

In [76]:
df["currency"] = df.item_price.str[0]
df


Unnamed: 0  order_id  quantity                              item_name  \
0              0         1         1           Chips and Fresh Tomato Salsa   
1              1         1         1                                   Izze   
2              2         1         1                       Nantucket Nectar   
3              3         1         1  Chips and Tomatillo-Green Chili Salsa   
4              4         2         2                           Chicken Bowl   
...          ...       ...       ...                                    ...   
4617        4617      1833         1                          Steak Burrito   
4618        4618      1833         1                          Steak Burrito   
4619        4619      1834         1                     Chicken Salad Bowl   
4620        4620      1834         1                     Chicken Salad Bowl   
4621        4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price currency  
0                                                   NaN        NaN      NaN  
1                                          [Clementine]     $3.39         $  
2                                               [Apple]     $3.39         $  
3                                                   NaN        NaN      NaN  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...    $16.98         $  
...                                                 ...        ...      ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...    $11.75         $  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...    $11.75         $  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...    $11.25         $  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...     $8.75         $  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     $8.75         $  

[4622 rows x 7 columns]

In [78]:
df.item_price = df.item_price.str.replace("$","")
df

Unnamed: 0  order_id  quantity                              item_name  \
0              0         1         1           Chips and Fresh Tomato Salsa   
1              1         1         1                                   Izze   
2              2         1         1                       Nantucket Nectar   
3              3         1         1  Chips and Tomatillo-Green Chili Salsa   
4              4         2         2                           Chicken Bowl   
...          ...       ...       ...                                    ...   
4617        4617      1833         1                          Steak Burrito   
4618        4618      1833         1                          Steak Burrito   
4619        4619      1834         1                     Chicken Salad Bowl   
4620        4620      1834         1                     Chicken Salad Bowl   
4621        4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price currency  
0                                                   NaN        NaN      NaN  
1                                          [Clementine]      3.39         $  
2                                               [Apple]      3.39         $  
3                                                   NaN        NaN      NaN  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...     16.98         $  
...                                                 ...        ...      ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...     11.75         $  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...     11.75         $  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     11.25         $  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...      8.75         $  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...      8.75         $  

[4622 rows x 7 columns]

In [79]:
df.item_price = df.item_price.fillna(0)
df

Unnamed: 0  order_id  quantity                              item_name  \
0              0         1         1           Chips and Fresh Tomato Salsa   
1              1         1         1                                   Izze   
2              2         1         1                       Nantucket Nectar   
3              3         1         1  Chips and Tomatillo-Green Chili Salsa   
4              4         2         2                           Chicken Bowl   
...          ...       ...       ...                                    ...   
4617        4617      1833         1                          Steak Burrito   
4618        4618      1833         1                          Steak Burrito   
4619        4619      1834         1                     Chicken Salad Bowl   
4620        4620      1834         1                     Chicken Salad Bowl   
4621        4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price currency  
0                                                   NaN          0      NaN  
1                                          [Clementine]      3.39         $  
2                                               [Apple]      3.39         $  
3                                                   NaN          0      NaN  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...     16.98         $  
...                                                 ...        ...      ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...     11.75         $  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...     11.75         $  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...     11.25         $  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...      8.75         $  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...      8.75         $  

[4622 rows x 7 columns]

In [86]:
df.item_price = df.item_price.astype("float")
df

Unnamed: 0  order_id  quantity                              item_name  \
0              0         1         1           Chips and Fresh Tomato Salsa   
1              1         1         1                                   Izze   
2              2         1         1                       Nantucket Nectar   
3              3         1         1  Chips and Tomatillo-Green Chili Salsa   
4              4         2         2                           Chicken Bowl   
...          ...       ...       ...                                    ...   
4617        4617      1833         1                          Steak Burrito   
4618        4618      1833         1                          Steak Burrito   
4619        4619      1834         1                     Chicken Salad Bowl   
4620        4620      1834         1                     Chicken Salad Bowl   
4621        4621      1834         1                     Chicken Salad Bowl   

                                     choice_description  item_price currency  
0                                                   NaN        0.00      NaN  
1                                          [Clementine]        3.39        $  
2                                               [Apple]        3.39        $  
3                                                   NaN        0.00      NaN  
4     [Tomatillo-Red Chili Salsa (Hot), [Black Beans...       16.98        $  
...                                                 ...         ...      ...  
4617  [Fresh Tomato Salsa, [Rice, Black Beans, Sour ...       11.75        $  
4618  [Fresh Tomato Salsa, [Rice, Sour Cream, Cheese...       11.75        $  
4619  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...       11.25        $  
4620  [Fresh Tomato Salsa, [Fajita Vegetables, Lettu...        8.75        $  
4621  [Fresh Tomato Salsa, [Fajita Vegetables, Pinto...        8.75        $  

[4622 rows x 7 columns]

In [98]:
(df.item_price*df.quantity).sum()

38998.020000000004

Le chiffre d'affaire est donc 38998.020000000004

## Q11 Quel est le revenu moyen par commande ?

In [91]:
df.groupby("item_name")["item_price"].mean()

item_name
6 Pack Soft Drink                         6.610185
Barbacoa Bowl                            10.187273
Barbacoa Burrito                          9.832418
Barbacoa Crispy Tacos                    10.928182
Barbacoa Salad Bowl                      10.640000
Barbacoa Soft Tacos                      10.018400
Bottled Water                             1.867654
Bowl                                     14.800000
Burrito                                   7.400000
Canned Soda                               1.320577
Canned Soft Drink                         1.457641
Carnitas Bowl                            10.833971
Carnitas Burrito                         10.132712
Carnitas Crispy Tacos                    11.137143
Carnitas Salad                            8.990000
Carnitas Salad Bowl                      11.056667
Carnitas Soft Tacos                       9.398500
Chicken Bowl                             10.113953
Chicken Burrito                          10.082857
Chicken Crispy Tacos 